scripts/code_style.py

```diff
@@ -163,14 +163,26 @@ def main() -> int:
                 + uncrustify_version + "' (Note: The only supported version" \
                 "is " + UNCRUSTIFY_SUPPORTED_VERSION + ")", file=STDOUT_UTF8)
 
-    src_files = get_src_files()
-
     parser = argparse.ArgumentParser()
-    parser.add_argument('-f', '--fix', action='store_true', \
-            help='modify source files to fix the code style')
+    parser.add_argument('-f', '--fix', action='store_true',
+                        help='modify source files to fix the code style')
+    # --files is almost useless: it only matters if there are no files
+    # ('code_style.py' without arguments checks all files known to Git,
+    # 'code_style.py --files' does nothing). 'code_style.py --files ...' is
+    # intended as a stable ("porcelain") way to restyle a possibly empty
+    # set of files.
+    parser.add_argument('--files', action='store_true',
+                        help='only check the specified files (default with non-option arguments)')
+    parser.add_argument('operands', nargs='*', metavar='FILE',
+                        help='files to check (if none: check files that are known to git)')
 
     args = parser.parse_args()
 
+    if args.files or args.operands:
+        src_files = args.operands
+    else:
+        src_files = get_src_files()
```

Comment:
The script continues to work on an invalid/missing passed file or a directory, looking for <missing_file>.uncrustify and if <missing_file>.uncrustify exists, there's a misleading message: `<missing_file> - OK.`.

Reply:
This is out of scope here. Generally speaking, the error handling in this script is fragile, which I've filed as https://github.com/Mbed-TLS/mbedtls/issues/6841